execute_until_build_succeeds.py

```
@@ -0,0 +1,137 @@
+                refined_unparsed = refine_code_based_on_suggestion(sut_content,
+                function_name,
+                additional_information,
+                knowledge_base_content,
+                test_file_path,
+                unit_testing_engine,
+                file_contents,
+                user_response,
+                unit_tests_first,
+                build_result)
+                # Getting the tool calls from the refined unit tests
+                toolsparsed = parse_function_calls_until_success(unit_tests, test_file_path)
```

Ruff (0.8.0)

`unit_tests` may be undefined, or defined from star imports

(F405)